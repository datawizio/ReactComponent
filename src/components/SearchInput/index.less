@import "../Input/index";
@import "../Button/index";