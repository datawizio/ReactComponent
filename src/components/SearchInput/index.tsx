import React from "react";

import { Input } from "antd";
import { SearchProps } from "antd/lib/input";

import "./index.less";

export interface SearchInputProps extends SearchProps {}

const SearchInput: React.FC<SearchInputProps> = props => {
  return <Input.Search {...props} allowClear />;
};

export default SearchInput;
