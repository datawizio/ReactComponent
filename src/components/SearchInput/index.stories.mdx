import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import SearchInput from "./index";

<Meta title="Components/Input/SearchInput" component={SearchInput} />

# SearchInput

Компонент предназначен для вывода поля для поиска.

Базовый компонент:
[**antd/input.search**](https://ant.design/components/input/).

## Basic

<Preview>
  <Story name="Basic">
    <SearchInput
      placeholder="Basic usage"
      onSearch={value => console.log(value)}
    />
  </Story>
</Preview>

## Small

<Preview>
  <Story name="Small">
    <SearchInput
      placeholder="Small size"
      size="small"
      onSearch={value => console.log(value)}
    />
  </Story>
</Preview>

## Large

<Preview>
  <Story name="Large">
    <SearchInput
      placeholder="Large size"
      size="large"
      onSearch={value => console.log(value)}
    />
  </Story>
</Preview>

## Loading

<Preview>
  <Story name="Loading">
    <SearchInput
      placeholder="Basic usage"
      onSearch={value => console.log(value)}
      loading
    />
  </Story>
</Preview>

## With button

<Preview>
  <Story name="With button">
    <SearchInput
      placeholder="With button usage"
      onSearch={value => console.log(value)}
      enterButton
    />
  </Story>
</Preview>

## With button - small

<Preview>
  <Story name="With button - small">
    <SearchInput
      placeholder="With button - small"
      size="small"
      enterButton
      onSearch={value => console.log(value)}
    />
  </Story>
</Preview>

## With button - large

<Preview>
  <Story name="With button - large">
    <SearchInput
      placeholder="With button - large"
      size="large"
      enterButton
      onSearch={value => console.log(value)}
    />
  </Story>
</Preview>

## With button - loading

<Preview>
  <Story name="With button, loading">
    <SearchInput
      placeholder="With button - loading"
      onSearch={value => console.log(value)}
      loading
      enterButton
    />
  </Story>
</Preview>

## Props

- [**antd/input.search**](https://ant.design/components/input/#Input.Search)

<Props of={SearchInput} />
