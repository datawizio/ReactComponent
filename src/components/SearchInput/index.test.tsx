import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import SearchInput from "./index";

const mockProps = {
  value: "val"
};

const setUp = (props?) => mount(<SearchInput {...props} />);

describe("SearchInput component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("SearchInput BackTop correctly", () => {
    expect(component).toMatchSnapshot();
  });
});
