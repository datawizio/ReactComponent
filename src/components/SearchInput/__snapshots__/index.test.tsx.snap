// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput component SearchInput BackTop correctly 1`] = `
<SearchInput
  value="val"
>
  <Search
    allowClear={true}
    enterButton={false}
    value="val"
  >
    <Input
      allowClear={true}
      className="ant-input-search"
      onChange={[Function]}
      onPressEnter={[Function]}
      prefixCls="ant-input"
      suffix={
        <ForwardRef(SearchOutlined)
          className="ant-input-search-icon"
          onClick={[Function]}
        />
      }
      type="text"
      value="val"
    >
      <ClearableLabeledInput
        allowClear={true}
        className="ant-input-search"
        element={
          <input
            className="ant-input ant-input-search"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            type="text"
            value="val"
          />
        }
        focused={false}
        handleReset={[Function]}
        inputType="input"
        onChange={[Function]}
        onPressEnter={[Function]}
        prefixCls="ant-input"
        suffix={
          <ForwardRef(SearchOutlined)
            className="ant-input-search-icon"
            onClick={[Function]}
          />
        }
        triggerFocus={[Function]}
        type="text"
        value="val"
      >
        <span
          className="ant-input-search ant-input-affix-wrapper ant-input-affix-wrapper-input-with-clear-btn"
          onMouseUp={[Function]}
        >
          <input
            className="ant-input"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            style={null}
            type="text"
            value="val"
          />
          <span
            className="ant-input-suffix"
          >
            <ForwardRef(CloseCircleFilled)
              className="ant-input-clear-icon"
              onClick={[Function]}
              role="button"
            >
              <AntdIcon
                className="ant-input-clear-icon"
                icon={
                  Object {
                    "icon": Object {
                      "attrs": Object {
                        "focusable": "false",
                        "viewBox": "64 64 896 896",
                      },
                      "children": Array [
                        Object {
                          "attrs": Object {
                            "d": "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z",
                          },
                          "tag": "path",
                        },
                      ],
                      "tag": "svg",
                    },
                    "name": "close-circle",
                    "theme": "filled",
                  }
                }
                onClick={[Function]}
                role="button"
              >
                <span
                  aria-label="close-circle"
                  className="anticon anticon-close-circle ant-input-clear-icon"
                  onClick={[Function]}
                  role="button"
                  tabIndex={-1}
                >
                  <IconReact
                    className=""
                    icon={
                      Object {
                        "icon": Object {
                          "attrs": Object {
                            "focusable": "false",
                            "viewBox": "64 64 896 896",
                          },
                          "children": Array [
                            Object {
                              "attrs": Object {
                                "d": "M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z",
                              },
                              "tag": "path",
                            },
                          ],
                          "tag": "svg",
                        },
                        "name": "close-circle",
                        "theme": "filled",
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="close-circle"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      key="svg-close-circle"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                        key="svg-close-circle-svg-0"
                      />
                    </svg>
                  </IconReact>
                </span>
              </AntdIcon>
            </ForwardRef(CloseCircleFilled)>
            <ForwardRef(SearchOutlined)
              className="ant-input-search-icon"
              key="searchIcon"
              onClick={[Function]}
            >
              <AntdIcon
                className="ant-input-search-icon"
                icon={
                  Object {
                    "icon": Object {
                      "attrs": Object {
                        "focusable": "false",
                        "viewBox": "64 64 896 896",
                      },
                      "children": Array [
                        Object {
                          "attrs": Object {
                            "d": "M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z",
                          },
                          "tag": "path",
                        },
                      ],
                      "tag": "svg",
                    },
                    "name": "search",
                    "theme": "outlined",
                  }
                }
                onClick={[Function]}
              >
                <span
                  aria-label="search"
                  className="anticon anticon-search ant-input-search-icon"
                  onClick={[Function]}
                  role="img"
                  tabIndex={-1}
                >
                  <IconReact
                    className=""
                    icon={
                      Object {
                        "icon": Object {
                          "attrs": Object {
                            "focusable": "false",
                            "viewBox": "64 64 896 896",
                          },
                          "children": Array [
                            Object {
                              "attrs": Object {
                                "d": "M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z",
                              },
                              "tag": "path",
                            },
                          ],
                          "tag": "svg",
                        },
                        "name": "search",
                        "theme": "outlined",
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      className=""
                      data-icon="search"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      key="svg-search"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                        key="svg-search-svg-0"
                      />
                    </svg>
                  </IconReact>
                </span>
              </AntdIcon>
            </ForwardRef(SearchOutlined)>
          </span>
        </span>
      </ClearableLabeledInput>
    </Input>
  </Search>
</SearchInput>
`;
